src/test/java/use_cases/PlayerGameInteractorTest.java

```diff
@@ -0,0 +1,221 @@
+     */
+    @Test
+    public void testCreateResponsePlayInvalidCard() {
+        // Give a new bogus Hand to p1 so to validate p1 has no valid Cards.
+        Hand newHand = new Hand(new ArrayList<>());
+        Card bogus1 = new Card("Test", "17");
+        Card bogus2 = new Card("Test", "18");
+        newHand.addCard(bogus1);
+        newHand.addCard(bogus2);
+        p1.setHand(newHand);
+        String chosenSuit = bogus1.getSuit();
+        String chosenValue = bogus1.getValue();
+        // Have p1 try to play bogus1.
+        PlayerGameRequestModel pgrm = new PlayerGameRequestModel("sol", chosenValue, chosenSuit, true, false, false);
+        interactor.createResponse(pgrm);
+        // Assert the turn has not changed, no winner, current Card is the same, none of the player's cards have changed.
+        assertEquals(p1, gameState.getCurrentPlayer());
+        assertFalse(gameState.getHasWinner());
+        assertEquals(firstCard, gameState.getCurrentCard());
+        for (Player p : game.getPlayers()) {
+            if (p.equals(p1)) {
+                assertEquals(2, p.getNumCards());
+            } else {
+                assertEquals(5, p.getNumCards());
+            }
+        }
+    }
+
+        /**
+         * Test createResponse when given a Player plays a valid Card and they become the winner.
+         */
+        @Test
+        public void testCreateResponsePlayCardWinner () {
+            // Give p1 a new Hand containing just one valid Card
+            Hand newHand = new Hand(new ArrayList<>());
+            newHand.addCard(p1sCard);
+            p1.setHand(newHand);
+            // Play the only Card in this Hand.
+            PlayerGameResponseModel response = interactor.createResponse(playCardReq);
+            // Assert the Game has a winner, p1 has no cards, p1 has 1 win 0 losses, while p3 has 1 loss and 0 win, current
+            // Card was the played Card.
+            assertTrue(response.getHasWinner());
+            assertEquals(0, p1.getNumCards());
+            HumanPlayer humanP1 = (HumanPlayer) p1;
+            assertEquals(1, humanP1.getStats()[0]);
+            assertEquals(0, humanP1.getStats()[1]);
+            HumanPlayer humanP3 = (HumanPlayer) p3;
+            assertEquals(1, humanP3.getStats()[1]);
+            assertEquals(0, humanP3.getStats()[0]);
+            assertEquals(p1sCard, gameState.getCurrentCard());
+        }
+
+        /**
+         * Test createResponse when a User/Player requests to skip a turn which they can.
+         */
+        @Test
+        public void testCreateResponseSkipTurnValid () {
+
+        }
+
+        /**
+         * Test createResponse when a User requests to skip a turn, but they cannot.
+         */
+        @Test
+        public void testCreateResponseSkipTurnInvalid () {
+            interactor.createResponse(skipReq);
+            // Nothing should have changed. Therefore, assert game was how it was before the response.
```

Comment:
Does the program let the user know that they cannot skip their turn? I was looking through the code but couldn't find if/where this occurs.

Reply:
When a Player clicks a button, either the action is done (they met the conditions) so the screen of the game changes to match what they did (in this case skipping a turn would show another players screen). If they didn't meet the conditions then the screen will stay as it is, no matter how many times they press the button, until they perform the right action.